src/main/java/com/avispl/symphony/dal/infrastructure/management/neat/pulse/NeatPulseCommunicator.java

```diff
@@ -233,6 +229,61 @@ public void stop() {
 		}
 	}
 
+	/**
+	 * Get list devices sensor information
+	 */
+	private void retrieveDeviceSensorInformation() {
+		try {
+			JsonNode response = this.doGet(String.format(NeatPulseCommand.LIST_DEVICE_SENSOR, this.getLogin()), JsonNode.class);
+			if (response != null && response.has(NeatPulseConstant.DATA)) {
+				for (JsonNode dataNode : response.get(NeatPulseConstant.DATA)) {
```

Comment:
is DATA node always going to be present? Would propose to either add a nullheck on dataNode, or use .at() otherwise